src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -73,6 +74,97 @@ get_pcidev(const xrt_core::device* device)
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+static query::sdm_sensor_info::result_type
+get_sdm_sensors(const xrt_core::device* device,
+                const query::sdm_sensor_info::sdr_req_type req_type,
+                const std::string path)
+{
+  auto pdev = get_pcidev(device);
+  query::sdm_sensor_info::result_type output;
+  query::sdm_sensor_info::data_type data;
+  std::vector<int> target_sysfs;
+  //All device sensors are stored in hwmon sysfs nodes with starts & ends as follows.
+  std::array<std::string, 5> start = {"curr", "in", "power", "temp", "fan"};
+  std::array<std::string, 5> end = {"label", "input", "max", "average", "highest"};
+  int max_end_types = end.size();
+
+  switch(req_type) {
+    case query::sdm_sensor_info::sdr_req_type::current:
+    case query::sdm_sensor_info::sdr_req_type::voltage:
+    case query::sdm_sensor_info::sdr_req_type::power:
+    case query::sdm_sensor_info::sdr_req_type::thermal: //temperature readings
+    case query::sdm_sensor_info::sdr_req_type::mechanical: //fan readings
+      target_sysfs.push_back((int)req_type);
+      break;
+    case query::sdm_sensor_info::sdr_req_type::electrical:
+      //electrical combines curr + volt + power readings
+      target_sysfs.push_back(0);
+      target_sysfs.push_back(1);
+      target_sysfs.push_back(2);
+      break;
+  }
+
+  //Iterate over hwmon sysfs nodes based on sensor request type.
+  for (const auto& sid : target_sysfs) {
```

Comment:
As an example of refactoring, maybe have a function that processes a single target_sysfs.